### PR TITLE
isbg.py : fix lock/pastuid file patterns when "imapproxyuser is None"

### PR DESCRIPTION
**Pull request by jimklimov**
Follow-up on PR #60

This should address at least part of issue #68 that relates to non-proxied mode of operation vs. single user only (I did not yet grasp the 16-byte ID suggestion).

### CODE REVIEW

**Comment by baldurmen** [comment]
I get the argument for the `%` and I would agree, but it does not need to be here at all.

If you look at your track files, they should look like this: `.isbg-track01d4a0289a2273b872e6d37d74a05c13`.

That md5 hash is generated by [this](https://github.com/isbg/isbg/blob/master/isbg.py#L266-L271) code. I was thus suggesting to try something like:

```
m = md5()
m.update(imaphost)
m.update(imapuser)
m.update(imapproxyuser)
m.update(repr(imapport))
res = hexof(m.digest())
pastuidsfile = pastuidsfile + res
```
I don't know if this works if `imapproxuser` is `None` (needs to be tested) and if it does not, you'd need a conditional statement, but that would respect the way it's done atm.

I know it's not pretty and having something readable by a human would be best, but I prefer to make that kind of modifs in a 2.0 version instead. Changing track files means you have to rerun everything  and that breaks user-space.

**Comment by jimklimov** [comment]
Looks interesting (and perhaps less prone to collisions due to ambiguous names)... but on the other hand, the plaintext version makes it easier to see which sessions are currently running ;)

**Comment by baldurmen** [comment]
I totally agree with you on the readability issue but to me, that's something that should be changed in a major version, not a minor one.

I see you PR as a way make ISBG not fail when using the `--imapproxyuser` parameter more than a general fix.

If you wish to modify the general behavior, I suggest you look at [this branch](https://github.com/isbg/isbg/tree/v.2.0).

**Comment by jimklimov** [comment]
Ok, I *finally* got your point by reading the few lines below where I was editing ;D

File: isbg.py [hunk 1]
```diff
@@ -304,13 +304,19 @@ def dehexof(x):
     m = md5()
     m.update(imaphost)
     m.update(imapuser)
+    if imapproxyuser is None:
+        m.update('')
```
Comment (baldurmen) [inline isbg.py]: I'm not sure why you run `m.update('')` instead of doing:

```
if imapproxyuser is not None:
    m.update(imapproxyuser)
```

Comment (jimklimov) [inline isbg.py]: I thought of it, but chose against for some reason... which can't readily remember now.
"Morning is smarter than the evening" as a saying goes; at least for backwards-compatibility with existing saved files, you're right - better not change the formula at all. :)

**Comment by baldurmen** [comment]
This is based on v.1.0.1 and will not be merged, since we are at v.2.0.0 atm and #86 will push us to v.2.1.0 with pretty significant code changes.

Chances are this problem has been solved since then anyway. Sorry for the bad communication on our side :(

**Pull request closed without merging**